### ISSUE TITLE
tree collapse/expand recursive

### ISSUE THREAD

**Issue by extrawurst** [extrawurst/gitui#708]
![Screenflick Movie 12](https://user-images.githubusercontent.com/776816/118396047-e7a4b400-b64d-11eb-9897-a320dc0e3940.gif)

**Comment by extrawurst** [comment]
implemented via 0e31d57

**Comment by nerdrew** [comment]
Usability thought: I assumed this didn't work because I tried it when a sub-file was selected. I expected the whole recursive tree to collapse when triggering "tree_collapse_recursive" with a sub-file selected. Nice feature though (once I figured out you need to have the top level directory selected) :+1:

**Comment by extrawurst** [comment]
> Usability thought: I assumed this didn't work because I tried it when a sub-file was selected. I expected the whole recursive tree to collapse when triggering "tree_collapse_recursive" with a sub-file selected. Nice feature though (once I figured out you need to have the top level directory selected) 👍

contrary to the gif above this feature is currently only implemented in the file tree view and not the status. it works by holding down `Shift` and pressing `Left`/`Right` on a folder node to collapse recursive or expand recursively

**Comment by nerdrew** [comment]
Yup. I got it working. But I was surprised that I couldn't collapse the whole tree when I had a file somewhere inside the tree selected. I needed to select the top level directory and then trigger collapse to have it work.

E.g. with this file selected (see screenshot), I couldn't collapse the tree.
![Screen Shot 2021-06-01 at 15 22 27](https://user-images.githubusercontent.com/10101/120397478-3d838800-c2ed-11eb-9c7b-aafa6f9e52de.png)
